arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -339,7 +339,7 @@ class ffmpeg_video_input::priv
       avformat_close_input(&this->f_format_context);
       this->f_format_context = nullptr;
     }
-
+    avcodec_close(this->f_video_encoding);
```

Comment:
This looks like the right thing to do, and the primary culprit of the memory leak.  However, for added safety and consistency for the other similar statements in this function, can you change it to
```c++
if (this->f_video_encoding)
{
  avcodec_close(this->f_video_encoding);
  this->f_video_encoding = nullptr;
}
```